File: Python/BOJ/15652.py
import sys

n, m = map(int, sys.stdin.readline().split())
lst = []


def dfs(start):
    if len(lst) == m:
        print(' '.join(map(str, lst)))
        return
    for i in range(1, n + 1):
        if (i >= start):
            lst.append(i)
            dfs(i)
            lst.pop()

dfs(1)

File: C++/BOJ/15652.cpp
#include<iostream>
#include<vector>

using namespace std;

int n, m;
vector<int> lst;

void    dfs(int start)
{
    if (lst.size() == m)
    {
        for (auto l : lst)
            cout<<l<<" ";
        cout<<"\n";
        return ;
    }
    for (int i = 1; i <= n; i++)
    {
        if (i < start)
            continue ;
        lst.push_back(i);
        dfs(i);
        lst.pop_back();
    }
}

int main()
{
    cin>>n>>m;
    dfs(1);
}